pulser-core/pulser/sequence/sequence.py

```diff
@@ -785,10 +786,223 @@ def check_retarget(ch_obj: Channel) -> bool:
+            strict: bool,
+        ) -> bool:
+            used_channels_new_device = list(channel_match.values())
+            if not reusable_channels and len(
+                set(used_channels_new_device)
+            ) < len(used_channels_new_device):
+                return False
+            for old_ch_name, new_ch_name in channel_match.items():
+                if check_channels_match(
+                    old_ch_name,
+                    all_channels_new_device[new_ch_name],
+                    active_eom_channels,
+                    strict,
+                ) != ("", ""):
+                    return False
+            return True
+
+        def raise_error_non_matching_channel(
+            reusable_channels: bool,
+            all_channels_new_device: dict[str, Channel],
+            active_eom_channels: list,
+            strict: bool,
+        ) -> None:
+            strict_error_message = ""
+            ch_match_err = ""
+            channel_match: dict[str, Any] = {}
+            for old_ch_name, old_ch_obj in self.declared_channels.items():
+                channel_match[old_ch_name] = None
+                base_msg = f"No match for channel {old_ch_name}"
+                # Find the corresponding channel on the new device
+                for new_ch_id, new_ch_obj in all_channels_new_device.items():
+                    if (
+                        not reusable_channels
+                        and new_ch_id in channel_match.values()
+                    ):
+                        # Channel already matched and can't be reused
+                        continue
+                    (ch_match_err_suffix, strict_error_message_suffix) = (
+                        check_channels_match(
+                            old_ch_name,
+                            new_ch_obj,
+                            active_eom_channels,
+                            strict,
+                        )
+                    )
+                    if (ch_match_err_suffix, strict_error_message_suffix) == (
+                        "",
+                        "",
+                    ):
+                        channel_match[old_ch_name] = new_ch_id
+                        # Found a match, clear match error msg for this channel
+                        if ch_match_err.startswith(base_msg):
+                            ch_match_err = ""
+                        if strict_error_message.startswith(base_msg):
+                            strict_error_message = ""
+                        break
+                    elif ch_match_err_suffix != "":
+                        ch_match_err = (
+                            ch_match_err
+                            if (
+                                "type, basis and addressing"
+                                in ch_match_err_suffix
+                                and ch_match_err
+                            )
+                            else base_msg + ch_match_err_suffix
+                        )
```

Comment:
Why do you keep the existing message only in this case. If I remember correctly, before we had somehting like
```suggestion
                        ch_match_err = (
                            ch_match_err
                            or (base_msg + ch_match_err_suffix)
                        )
```
What's wrong with this?

Reply:
My bad, I think what you are suggesting is better than what I did at the moment, I forgot how it was before when writing this 😅